pkg/descheduler/evictions/evictions.go

```diff
@@ -166,11 +170,11 @@ func (pe *PodEvictor) EvictPod(ctx context.Context, pod *v1.Pod, opts EvictOptio
 		klog.V(1).InfoS("Evicted pod in dry run mode", "pod", klog.KObj(pod), "reason", opts.Reason, "strategy", strategy, "node", pod.Spec.NodeName)
 	} else {
 		klog.V(1).InfoS("Evicted pod", "pod", klog.KObj(pod), "reason", opts.Reason, "strategy", strategy, "node", pod.Spec.NodeName)
-		eventBroadcaster := record.NewBroadcaster()
-		eventBroadcaster.StartStructuredLogging(3)
-		eventBroadcaster.StartRecordingToSink(&clientcorev1.EventSinkImpl{Interface: pe.client.CoreV1().Events(pod.Namespace)})
-		r := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "sigs.k8s.io.descheduler"})
-		r.Event(pod, v1.EventTypeNormal, "Descheduled", fmt.Sprintf("pod evicted by sigs.k8s.io/descheduler%s", opts.Reason))
+		reason := opts.Reason
+		if len(reason) == 0 {
+			reason = "NotSet"
```

Comment:
We are not setting Reasons yet, but if we try to emit events with Event and a empty Reason, it fails. I am leaving this here but we can remove when we start setting them in all strategies

Reply:
we used to [set](https://github.com/kubernetes-sigs/descheduler/blob/v0.24.1/pkg/descheduler/evictions/evictions.go#L125) reason to strategy name if it wasn't set.

Reply:
any reason we stopped doing that?

Reply:
Ok, did something similar now